OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -75,12 +75,10 @@ public List<CPos> FindUnitPath(CPos source, CPos target, Actor self, Actor ignor
 			if (domainIndex != null && !domainIndex.IsPassable(source, target, locomotor))
 				return EmptyPath;
 
-			var distance = source - target;
-			var canMoveFreely = locomotor.CanMoveFreelyInto(self, target, check, null);
-			if (distance.LengthSquared < 3 && !canMoveFreely)
```

Comment:
Can you explain why this has been removed? I don't remember the history off the top of my head, but i'm fairly sure this was working around a specific bug, so removing it may regress things.

Reply:
It was added for the case when a path between two neighbor cells and only the direct path existed, which caused the bidirectional pathfinder to fail.

Reply:
The above part basicaly skips the pathfinding logic for adjacent cell movement (< 3) any it was assumed "if that cell is free, we can always directly move into it". Which is no longer true due to this prs behavior implementation. Diagonal can still be blocked requiring a proper path check.

Reply:
I mixed up this code block with the one below, this code was added as a optimization, instead of running the whole astar setup for adjacent cell movement (< 3) we can return early if no such path is available.